packages/docusaurus-theme-classic/src/theme/CodeBlock/WordWrapButton/index.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useCallback, useState, useEffect} from 'react';
+import clsx from 'clsx';
+import type {Props} from '@theme/CodeBlock/WordWrapButton';
+
+import styles from './styles.module.css';
+
+export default function WordWrapButton({
+  codeBlockRef,
+  className,
+}: Props): JSX.Element | null {
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isCodeScrollable, setIsCodeScrollable] = useState<boolean>(false);
+  const toggleWordWrapCode = useCallback(() => {
+    setIsEnabled((value) => !value);
+    const codeElement = (codeBlockRef.current as HTMLPreElement).querySelector(
+      'code',
+    ) as HTMLElement;
+    codeElement.classList.toggle(styles.codeWithWordWrap as string);
+  }, [codeBlockRef]);
+
+  useEffect(() => {
+    if (codeBlockRef.current) {
+      setIsCodeScrollable(
+        codeBlockRef.current.scrollWidth > codeBlockRef.current.clientWidth,
+      );
+    }
+  }, [codeBlockRef]);
```

Comment:
Should this also listen to screen resizes?

Reply:
Good idea, at least when changing the orientation of a device it will definitely come in handy.